tests/cephfs/cephfs_fscrypt/test_fscrypt_basic.py

```diff
@@ -0,0 +1,530 @@
+    FScrypt functional tests - Polarion TC CEPH-83607378
+    ------------------------   -------------------------
+    Type - Sanity / Lifecycle
+    Workflow1:
+    1.Create FS volume and subvolumes across default and non-default groups, 2 in default and 1 in non-default
+    2.Mount subvolumes with kernel client
+    3.Create 2 test directories in each subvolumes
+    4.Perform fscrypt setup on mount point of one subvolume from each group
+    5.Perform fscrypt encrypt on 2 test directories of subvolumes where fscrypt setup done
+    6.In test directories across all subvolumes, add directories with depth as 10 and breadth as 5
+    and 2 files in each dir
+    7.Run FIO on each file in continuous mode until execution run time in background
+    8.On subvolume mount points where fscrypt is setup, perform below ops,
+        a.fscrypt encrypt ( this internally creates protector and policy and attaches to encryption path)
+        b.Validate file and directories(store file and directory names before lock) - name and content
+        c.fscrypt lock and validate file and dir names are encrypted, file contents are encrypted
+        d.fscrypt unlock and validate file and dir names are readable, file contents are readable.
+        e.fscrypt purge - To remove encryption on test directories
+
+    Workflow2:
+    Verify fscrypt encrypt not supported for non-empty directory path
+
+    Workflow3:
+    Verify fscrypt encrypt lock does not encrypt metadata of files including, size, timestamp,permissions and
+    extended sttributes
+
+    Clean Up:
+        Unmount subvolumes
+        Remove subvolumes and subvolumegroup
+        Remove FS volume if created
+    """
+    try:
+        test_data = kw.get("test_data")
+        fs_util = FsUtilsv1(ceph_cluster, test_data=test_data)
+        cephfs_common_utils = CephFSCommonUtils(ceph_cluster)
+        erasure = (
+            FsUtilsv1.get_custom_config_value(test_data, "erasure")
+            if test_data
+            else False
+        )
+        fscrypt_util = FscryptUtils(ceph_cluster)
+
+        config = kw.get("config")
+        clients = ceph_cluster.get_ceph_objects("client")
+        if len(clients) < 2:
+            log.info(
+                "This test requires minimum 2 client nodes.This has only %s clients",
+                len(clients),
+            )
+            return 1
+
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        fs_util.auth_list(clients)
+        default_fs = config.get("fs_name", "cephfs")
+        default_fs = default_fs if not erasure else "cephfs-ec"
+        cleanup = config.get("cleanup", 1)
+        client = clients[0]
+        client1 = clients[1]
+        log.info("checking Pre-requisites")
+        log.info("Verify Cluster is healthy before test")
+        if cephfs_common_utils.wait_for_healthy_ceph(client, 300):
+            assert False, "Cluster health is not OK even after waiting for 300secs"
```

Comment:
Please avoid using `assert` statements. You can return or break the implementation. An unhandled `assert` would stop the execution workflow. This has an impact when we are running in parallel.